src/compiler/transformers/classFields.ts

```diff
@@ -632,7 +723,21 @@ namespace ts {
             // From ES6 specification:
             //      HasLexicalDeclaration (N) : Determines if the argument identifier has a binding in this environment record that was created using
             //                                  a lexical declaration such as a LexicalDeclaration or a ClassDeclaration.
-            const staticProperties = getProperties(node, /*requireInitializer*/ true, /*isStatic*/ true);
+            let staticProperties = getProperties(node, /*requireInitializer*/ true, /*isStatic*/ true);
+
+            if (languageVersion === ScriptTarget.ESNext) {
+                // We can't initialize static private properties after the class as the are not in the lexical scope
+                // so we rely on standard init even under useDefineForClassFields:false

```

Comment:
Actually since we decided emit semantics are broken if we do this, I will just remove this code and emit invalid code under `target:esnext & useDefineForClassFields:false`